shell/platform/android/io/flutter/view/TextureRegistry.java

```diff
@@ -57,7 +66,45 @@ interface TextureEntry {
     void release();
   }
 
+  interface ImageProducerTextureEntry extends TextureEntry {
+    /**
+     * Retrieve the last Image produced. Drops all previously produced images.
+     *
+     * <p>NOTE: Caller must call Image.close() on returned image.
+     *
+     * @return Image or null.
+     */
+    public Image acquireLatestImage();
+  }
+
+  /** A registry entry for texture updated by rendering into a Surface. */
+  @Keep
+  interface RawSurfaceTextureEntry extends TextureEntry {
+    /** @return The identity of this texture. */
+    long id();
+
+    /** Deregisters and releases all resources . */
+    void release();
+
+    /** Specify the size of this texture. */
+    void setSize(int width, int height);
+
+    /** @return The currently specified width. */
+    int getWidth();
+
+    /** @return The currently specified height. */
+    int getHeight();
+
+    /**
+     * Get a Surface that can be used to update the texture contents.
+     *
+     * @return a Surface to use for a drawing target for various APIs.
+     */
+    Surface getSurface();
```

Comment:
What happens if I call this multiple times? I think by convention, it will always return the same surface since its a `getFoo` method. Or does that work differently with the ImageReader backend approach?

Basically, to use this, do I need to occasionally handle grabbing a new surface object?

Reply:
Or is it "if you don't resize then this is always the same surface, but if you do resize, then its a new surface?"